File: backend-mvc/src/main/java/sba/group3/backendmvc/repository/user/UserRepository.java
package sba.group3.backendmvc.repository.user;

import sba.group3.backendmvc.entity.user.User;
import sba.group3.backendmvc.repository.BaseRepository;

import java.util.Optional;
import java.util.UUID;

public interface UserRepository extends BaseRepository<User, UUID> {
    Optional<User> findByUsername(String username);
}

File: chat-backend/src/main/kotlin/com/sba301/chatbackend/repository/UserRepository.kt
package com.sba301.chatbackend.repository

import com.sba301.chatbackend.entity.User
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import org.springframework.stereotype.Repository
import reactor.core.publisher.Mono

@Repository
interface UserRepository : ReactiveMongoRepository<User, String> {
    fun findByUsername(username: String): Mono<User>
    fun existsByUsername(username: String): Mono<Boolean>
}
